File: Data Structures & Algorithms/maximum-subarray/submission-2.go
func maxSubArray(nums []int) int {
	ans := nums[len(nums)-1]
	rs := ans
	for i:=len(nums)-2;i>=0;i-=1{
		cur := nums[i]
		if rs < 0{
			rs = cur
		}else{
			rs = cur + rs
		}
		ans = max(ans, rs)
	}
	return ans
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()
        ans = 0

        for i in nums:
            s.add(i)
        
        for i in nums:
            prev = i - 1
            if prev in s:
                continue
            
            t = i
            count = 0
            while t in s:
                count += 1
                ans = max(ans, count)
                t += 1
        return ans
                
        